rfcs/text/0011_global_search.md

```diff
@@ -0,0 +1,693 @@
+only uses the last one (For example in the UI, a debounced `find` triggered on end-user keypress: On every new keypress,
+a new search is performed and the previous, potentially still executing, `find` call will still be executed), it can't
+cancel the previous active search, resulting on potential useless http calls and resources consumption.
+
+We could add an optional cancellation signal or observable in `GlobalSearchOptions` and propagate it to result providers.
+That way `find` calls could be aborted.
+
+However result providers would still have to manually handles this signal to cancel any http call or 
+other asynchronous task that could be pending. 
+
+Notes:
+- As this can be implemented with an additional option, this would be a non-breaking change and so could be done at 
+  a later time (even if it means that existing result provider would not handles this option until they implements it).
+
+## The GlobalSearch service could be provided as a plugin instead of a core service
+
+The GlobalSearch API could be provided and exposed from a plugin instead of `core`.
+
+Pros:
+- Less `core` API exposure
+
+Cons:
+- We know our initial consumer of this API is going to use it from/for the chrome header component, and because this component 
+  is in `core`, that would mean creating a bridge of some kind to be able to use the service from core, as plugin APIs
+  are not usable here.
+- The platform team is going to provide the base result providers for SO and application search results, so having the GS
+  service in a plugin would mean creating yet another plugin for these providers instead of having them self contained
+  in core. 
+- We are probably going to be adding internal SO APIs for the SO result provider, as current APIs are not sufficient to
+  search for multiple type of SO at the same time. It would be better if these APIs were kept internal, which would
+  not be possible if the SO provider is in a plugin instead of core.
+
+# Adoption strategy
+
+The `globalSearch` service is a new feature provided by the `core` API. Also, the base providers
+used to search for saved objects and applications will be implemented by the platform team, meaning
+that by default, plugin developers won't have to do anything.
+
+Plugins that wish to expose additional result providers will easily be able to do so by using the exposed APIs and
+documentation.
+
+# How we teach this
+
+This follows the same patterns we have used for other Core APIs: Observables subscriptions, etc.
+
+This should be taught using the same channels we've leveraged for other Kibana Platform APIs, API documentation mostly.
+
+# Unresolved questions
+
+## terminology / naming
+
+Are the current types, services and api names acceptable:
+   - `GlobalSearch` ts prefix
+   - `core.globalSearch` / `GlobalSearchService`
+   - `GlobalSearchResultProvider`
+   - `core.globalSearch.find`
```

Comment:
@stacey-gammon I know you have an opinion on this one 😄. Please go on.

Reply:
Haha, just today I was having a conversation with @elastic-jb and he also was a bit confused by the "search" terminology since it's so often meant to mean searching Elasticsearch data indices.  On the flip side, it matches "Enterprise search".  They are also adding an App called "App Search" to Kibana I think.

I like the `find` name in there.  Makes me think of the mac finder which this feels more similar to, as opposed to data searching/querying.  Very happy to see `DataSources` terminology gone, thank you! 

Ultimately, I'm okay with any of these though.